socorro/cron/jobs/daily_url.py

```diff
@@ -2,103 +2,329 @@
+    config.logger.debug(
+        "day = %s; now = %s; yesterday = %s",
+        day,
+        now,
+        yesterday
+    )
+    prod_phrase = ''
+    try:
+        if config.product != '':
+            if ',' in config.product:
+                prod_list = [x.strip() for x in config.product.split(',')]
+                prod_phrase = (
+                    "and r.product in ('%s')" % "','".join(prod_list)
+                )
+            else:
+                prod_phrase = "and r.product = '%s'" % config.product
+    except Exception, x:
+        config.logger.error(repr(x), exc_info=True)
+    ver_phrase = ''
+    try:
+        if config.version != '':
+            if ',' in config.product:
+                ver_list = [v.strip() for v in config.version.split(',')]
+                ver_phrase = ("and r.version in ('%s')" % "','".join(ver_list))
+            else:
+                ver_phrase = "and r.version = '%s'" % config.version
+    except Exception, x:
+        config.logger.error(repr(x), exc_info=True)
+
+    return DotDict({
+        'now_str': now_str,
+        'yesterday_str': yesterday_str,
+        'prod_phrase': prod_phrase,
+        'ver_phrase': ver_phrase
+    })
+
+
+#------------------------------------------------------------------------------
+@contextlib.contextmanager
+def gzipped_csv_files(config, day, gzip=gzip, csv=csv):
+    private_out_filename = (
+        "%s-crashdata.csv.gz" % day.strftime('%Y%m%d')
+    )
+    private_out_pathname = os.path.join(
+        config.output_path,
+        private_out_filename
+    )
+    config.private_out_pathname = private_out_pathname
+    private_gzip_file_handle = gzip.open(private_out_pathname, "w")
+    private_csv_file_handle = csv.writer(
+        private_gzip_file_handle,
+        delimiter='\t',
+        lineterminator='\n'
+    )
+
+    pubic_out_filename = ("%s-pub-crashdata.csv.gz"
+                          % day.strftime('%Y%m%d'))
+    public_out_pathname = None
+    public_out_directory = config.get('public_output_path')
+    public_gzip_file_handle = None
+    public_csv_file_handle = None
+    if public_out_directory:
+        public_out_pathname = os.path.join(
+            public_out_directory,
+            pubic_out_filename
+        )
+        public_gzip_file_handle = gzip.open(public_out_pathname, "w")
+        public_csv_file_handle = csv.writer(
+            public_gzip_file_handle,
+            delimiter='\t',
+            lineterminator='\n'
+        )
+    else:
+        config.logger.info("Will not create public (bowdlerized) gzip file")
+    config.public_out_pathname = public_out_pathname
+    yield (private_csv_file_handle, public_csv_file_handle)
+    private_gzip_file_handle.close()
+    if public_gzip_file_handle:
+        public_gzip_file_handle.close()
+
+
+#------------------------------------------------------------------------------
+linux_line_re = re.compile(
+    r'0\.0\.0 [lL]inux.+[lL]inux$|[0-9.]+.+(i586|i686|sun4u|i86pc|x86_64)?'
```

Comment:
this is code relocated from the IdCache class - as that class is deprecated, I copied this part literally.  if it has a problem, we ought to deal with it in a separate Bug/PR.